src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,384 @@
+        public IntPtr GetOrCreateComInterfaceForObject(object instance, CreateComInterfaceFlags flags)
+        {
+            IntPtr ptr;
+            if (!TryGetOrCreateComInterfaceForObjectInternal(this, instance, flags, out ptr))
+                throw new ArgumentException();
+
+            return ptr;
+        }
+
+        /// <summary>
+        /// Create a COM representation of the supplied object that can be passed to a non-managed environment.
+        /// </summary>
+        /// <param name="impl">The <see cref="ComWrappers" /> implementation to use when creating the COM representation.</param>
+        /// <param name="instance">The managed object to expose outside the .NET runtime.</param>
+        /// <param name="flags">Flags used to configure the generated interface.</param>
+        /// <param name="retValue">The generated COM interface that can be passed outside the .NET runtime or IntPtr.Zero if it could not be created.</param>
+        /// <returns>Returns <c>true</c> if a COM representation could be created, <c>false</c> otherwise</returns>
+        /// <remarks>
+        /// If <paramref name="impl" /> is <c>null</c>, the global instance (if registered) will be used.
+        /// </remarks>
+        private static bool TryGetOrCreateComInterfaceForObjectInternal(ComWrappers? impl, object instance, CreateComInterfaceFlags flags, out IntPtr retValue)
+        {
+            if (instance == null)
+                throw new ArgumentNullException(nameof(instance));
+
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Compute the desired Vtable for <paramref name="obj"/> respecting the values of <paramref name="flags"/>.
+        /// </summary>
+        /// <param name="obj">Target of the returned Vtables.</param>
+        /// <param name="flags">Flags used to compute Vtables.</param>
+        /// <param name="count">The number of elements contained in the returned memory.</param>
+        /// <returns><see cref="ComInterfaceEntry" /> pointer containing memory for all COM interface entries.</returns>
+        /// <remarks>
+        /// All memory returned from this function must either be unmanaged memory, pinned managed memory, or have been
+        /// allocated with the <see cref="System.Runtime.CompilerServices.RuntimeHelpers.AllocateTypeAssociatedMemory(Type, int)"/> API.
+        ///
+        /// If the interface entries cannot be created and a negative <paramref name="count" /> or <code>null</code> and a non-zero <paramref name="count" /> are returned,
+        /// the call to <see cref="ComWrappers.GetOrCreateComInterfaceForObject(object, CreateComInterfaceFlags)"/> will throw a <see cref="System.ArgumentException"/>.
+        /// </remarks>
+        protected unsafe abstract ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count);
+
+        // Called by the runtime to execute the abstract instance function
```

Comment:
There is nothing calling this method from the runtime. Do you plan to hook it up in this PR?

I would be ok with merging a PR that just adds NYI stubs for everything, but it should be self-consistent and not contain dead code.

Reply:
I really want start discussion again with this PR. I plan to go with implementing all stuff in C#.

Can I borrow current implementation and replace `QCall` with dummy methods which will throw `NotImplmentedException`. Does that seems like reasonable intermediate PR? Will this tick at least one checkbox in the #306 ?

Reply:
I agree that it is a good idea to submit smaller intermediate PRs to make progress. The intermediate PRs should be self-consitent.

Future proofing, like adding unrechable private methods that may be used in future, rarely works well. The private methods typically end up being reworked once they get actually used.